### PR TITLE
IN_parser_fix_historical

### PR DESCRIPTION
**Pull request by mathilde-daugy**
## Description

Parser was unstable because the link to the CEA files changes depending on target_datetime.
The new parser now tries different vesions of the target link and if this fails, it will try to collect data from renewablesindia.in
The reports made available there are the same as on CEA but there are delayed compared to the daily CEA reports and the time series is not complete either. 

### Preview
image

### CODE REVIEW

**Review by VIKTORVAV99 — COMMENTED** [review]
Are we modifying the datetimes at any point or just returning one of multiple datapoints?
I can seem to get to to return anything but the datapoint for 18:30:00+00:00 UTC (01:00 local time) (tried with multiple dates)

**Comment by mathilde-daugy** [comment]
> Are we modifying the datetimes at any point or just returning one of multiple datapoints? I can seem to get to to return anything but the datapoint for 18:30:00+00:00 UTC (01:00 local time) (tried with multiple dates and times)

This parser returns only one data point per day. After investigating for a long time i found that this was the best source for India. We will be developing an estimation model to create an hourly profile later using demand data.

File: parsers/IN.py [hunk 1]
```diff
@@ -283,35 +312,86 @@ def fetch_cea_production(
...
+        "{date.day}_{date:%b_%Y}_Daily_Report.xlsx",
+        "{date.day}_{date.month}_{date.year}_Daily_Report.xlsx",
+        "DailyRE{date:%d_%m_%Y}.xlsx",
+        "DailyRE{date:%d%m%Y}.xlsx",
+        "{date.day}_{date:%B__%Y}.xlsx",
+        "{date.day}_{date:%B_%Y}.xlsx",
+        "{date.day}_{date:%B_%Y}_Daily_Report.xlsx",
+        "{date.day}_{date:%b__%Y}.xlsx",
+        "{date.day}_{date:%b_%Y}.xlsx",
+    ]
```
[the remark this answers is not in the document]
Comment (mathilde-daugy) [reply parsers/IN.py:329]: I've tried to put the most relevant links at the beginning of the list. It looks like the file naming is ok in 2023 but before that there is no consistency...  Idk if there is a faster way around it ?

**Comment by mathilde-daugy** [comment]
Sorry @VIKTORVAV99, I'm changing this around quite a bit but i finally found a way to make the parser more stable. I was able to find the query with all the links to the data files! Sorry for the bother

**Comment by VIKTORVAV99** [comment]
> Sorry @VIKTORVAV99, I'm changing this around quite a bit but i finally found a way to make the parser more stable. I was able to find the query with all the links to the data files! Sorry for the bother

No problem! I'll take a look at the updated version right now!

**Review by VIKTORVAV99 — APPROVED** [review]
Looks good and seems to be more stable and faster than the old version that looped over the diffrent links.
(Should be more efficent as well)

**Pull request merged**